src/Analyzers/Core/CodeFixes/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsCodeFixProvider.cs

```diff
@@ -29,7 +29,10 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
             CodeAction.Create(
                 title,
                 cancellationToken => RemoveUnnecessaryImportsAsync(context.Document, cancellationToken),
-                title),
+                title,
+                // If the user is on a using/import that is marked as unnecessary, then we want to make sure that this
+                // code action is preferred over virtually any others that could be located at that position.
+                priority: CodeActionPriority.High),
```

Comment:
Note: the attributes solve this specific issue.  But i wanted to ensure that we wouldn't have the problem with other fixers in the future.